lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -928,6 +933,19 @@ class EmitterBase {
     emitLocationImpl(
         getLocationInfoAsString(ops, state.options.locationInfoStyle));
     setPendingNewline();
+    auto *ctxt = state.designOp.getContext();
+    // The column of the stream is incorrect, we can drop it here.
+    auto verilogLoc = DictionaryAttr::get(
+        ctxt,
+        {NamedAttribute(StringAttr::get(ctxt, "col"),
+                        StringAttr::get(ctxt, Twine(state.os.getColumn()))),
+         NamedAttribute(StringAttr::get(ctxt, "line"),
+                        StringAttr::get(ctxt, Twine(state.os.getLine() + 1))),
+         NamedAttribute(StringAttr::get(ctxt, "file"),
+                        StringAttr::get(ctxt, state.fileName))});
+
+    for (auto *op : ops)
+      op->setAttr(StringAttr::get(ctxt, "verilogLocation"), verilogLoc);
   }
```

Comment:
I think this is great start, eventually we will want to get more precise location by adding callback to PrettyPrinter but as a starting point this looks good to me. Note that op could be emitted multiple times (e.g. constant, extract and few others..) so we need to find a way to deal with that.

Reply:
Thanks Hideto for the idea of the `formatted_raw_ostream`, I am going to try out the callback to PrettyPrinter next, this was just an experiment and at least the line number seemed to be correct for the cases I looked at.